pkg/certwatcher/certwatcher.go

```diff
@@ -139,6 +163,8 @@ func (cw *CertWatcher) Watch() {
 			}
 
 			log.Error(err, "certificate watch error")
+		case <-watcherHealthTimer.C:
+			cw.ensureAllFilesAreWatched()
```

Comment:
Permanently polling once per second seems pretty aggrsive. Can we not watch the dir intead if the file gets removed and re-add the filewatch once it exists again?

Reply:
sidenote: wondering at this point if we should generally avoid relying of fs events, and rather say poll every N seconds and reload the certificates if the hash changed or something?

Reply:
We could also make the polling configurable, and default it to something reasonable where both old (loaded) and new certificates are supposed to be valid

Reply:
> Can we not watch the dir intead if the file gets removed and re-add the filewatch once it exists again?

It does not watch the dir, it just tries to reinstall the notify watch that fails if the file is not present. 
So technically, it is the same as "waiting until the file exists again" as the first step of `addWatch` is `os.Lstat(name)`. 

The other option is to watch the directory events instead of file events and parse all events for the file names but it:
1. Increases the number of events to parse and handle
2. Does not solve the case when the directory would be removed and recreated again (or have mount issues, whatever).

Reply:
> sidenote: wondering at this point if we should generally avoid relying of fs events, and rather say poll every N seconds and reload the certificates if the hash changed or something?

It makes sense, and in some of our internal projects, we rely on this behaviour instead. The downside is that you need to find a balance between polling interval, certificate expiration time, and number of file reads in this case; so ideally this param should also be configurable across all usages of certwatcher (webhook server and metrics server at least)

Reply:
Thank you for your suggestions.
@alvaroaleman @vincepri I updated the PR with full removal of fsnotify from certwatcher and having a simple read/cache/repeat ticker loop. Please check and leave your opinions.